### PR TITLE
feat: upgrade Gateway API to v1beta1

### PR DESCRIPTION
**Pull request by horjulf**
Upgrades dependencies to latest, in specific Gateway API to release `0.8.1`.

`v1alpha2` is no longer supported, so this drops supports for that and aligns with usptream.

```
v1alpha2 of Gateway, GatewayClass, and HTTPRoute is no longer served
v1alpha2 of ReferenceGrant is deprecated
v1beta1 is now the storage version for ReferenceGrant
```

https://github.com/kubernetes-sigs/gateway-api/releases/tag/v0.8.1

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,69 +1,73 @@
...
+	github.com/miekg/dns v1.1.55
 	github.com/nginxinc/kubernetes-ingress v1.12.5
-	k8s.io/api v0.26.3
-	k8s.io/apimachinery v0.26.3
-	k8s.io/client-go v0.26.3
-	sigs.k8s.io/gateway-api v0.4.3
+	k8s.io/api v0.28.1
+	k8s.io/apimachinery v0.28.1
+	k8s.io/client-go v0.28.1
+	sigs.k8s.io/gateway-api v0.8.0
```
Comment (larivierec) [inline go.mod:13]: should probably bump this since it has recently been updated

Comment (horjulf) [reply go.mod]: @larivierec Thanks, updated.

**Comment by larivierec** [comment]
@networkop  hey there, apologies for pinging you is there anything that can be done to get traction on this PR?

**Comment by networkop** [comment]
@horjulf @larivierec some of the tests are failing, can you PTAL?

**Comment by larivierec** [comment]
> @horjulf @larivierec some of the tests are failing, can you PTAL?

Yup, I'll take a look :)

**Comment by larivierec** [comment]
@horjulf, probably going to need to bump this to 1.21.3 https://github.com/ori-edge/k8s_gateway/blob/master/Dockerfile#L1

**Comment by horjulf** [comment]
@networkop Golang version updated to 1.21, that should fix the tests.

**Comment by networkop** [comment]
looks like the linter is still failing but all tests pass, so I'll merge it.

**Pull request merged**